package.json

```diff
@@ -21,30 +21,31 @@
     "tag manager",
     "SDK"
   ],
-  "main": "./src/reactor.js",
+  "main": "./dist/reactor-node.js",
```

Comment:
That was just a mistake on my part. It had intended to point `main` at `lib/index.js` for node. 

For browser-based clients:
I had assumed they would use the bundled version in `dist/reactor-browser.min.js`. Your comment makes it sound like you'd expect them to use `lib/index.js`, just like the node users would. Which makes me wonder why the bundled version in `dist/` even needs to exist. Though I do remember you mentioning that you weren't sure there was even a need for the bundle in modern usage; maybe this is why?

Reply:
On the subject of browser-based clients: I'm thinking that maybe (1) `lib/index.js` is those who intend to bundle their JavaScript + our SDK, and (2) `dist/reactor-browser.min.js` is for the dwindling population of those who intend to just include the SDK with `<script src="dist/reactor-browser.min.js"></script>`?

Reply:
Yes, what you've stated is correct. I think the problem you'll run into is that it looks like `lib/index.js` requires `node-fetch`. `node-fetch` is node-only, so if someone tried to bundle the SDK for use in a browser, it would throw errors when running in the browser because things like [stream, http, zlib, etc](https://github.com/bitinn/node-fetch/blob/master/src/index.js#L10-L14) are things that node provides and the browser does not.

The way I think I would probably handle this is have a `nodeIndex.js` and a `browserIndex.js`.`nodeIndex.js`. Each of those files would manage getting a `fetch` implementation for their specific environment, then passing the `fetch` implementation into an SDK factory of sorts. In the case of the browser environment, you'd just use `window.fetch`. In the case of node, you would use `node-fetch`. For the SDK factory, I would rename what is currently `index.js` to something like `reactorFactory.js` and have it export a function that accepts the `fetch` implementation and then return your Reactor class, which would use the `fetch` implementation that was passed in.

At that point, you need to point consumers targeting the browser to `browserIndex.js` and consumers targeting node to `nodeIndex.js`. I think that's where the [`main` and `browser` package.json fields](http://2ality.com/2017/04/setting-up-multi-platform-packages.html#browser-browser-specific-code) come into play. Here's the [official documentation](https://docs.npmjs.com/files/package.json#browser) for the `browser` package.json field.

Does that make sense? Again, I haven't actually ever worked on a library that targets both node and browser, so you'll probably want to test everything out as though you were a consumer.

Reply:
Waiiiiit. Nevermind. I just noticed that `node-fetch` [uses a `browser` field in its package.json](https://github.com/bitinn/node-fetch/blob/432c9b01ea71cdf0513258ed128688437796e54a/package.json#L5-L6) to point to a browser-compliant version when bundling for the browser. In other words, they take care of that for you of handling the node vs browser differences for you. That's really great. That's why your `dist/reactor-browser.min.js` doesn't try to require node-specific packages. You probably already knew this, but I didn't. Anyway, it doesn't look like you'll run into the problem I thought you'd run into, so you can disregard what I was saying.